Formula/haskell-language-server.rb

```diff
@@ -63,11 +78,11 @@ def caveats
       f :: Int -> Int
     EOS
 
-    %w[ghc@8.6 ghc@8.8 ghc].each do |ghc_str|
-      ghc = Formula[ghc_str]
-
+    ghcs.each do |ghc|
       assert_match "Completed (1 file worked, 1 file failed)",
-        shell_output("PATH=#{ghc.bin}:$PATH #{bin}/haskell-language-server-#{ghc.version} #{testpath}/*.hs 2>&1", 1)
+        with_env(PATH: "#{ghc.bin}:#{ENV["PATH"]}") do
+          shell_output("#{bin}/haskell-language-server-#{ghc.version.major_minor} #{testpath}/*.hs 2>&1", 1)
+        end
```

Comment:
I don't think `with_env` block will return anything.

If this is wanted, then may need to use a variable
```rb
ghcs.each do |ghc|
  output = nil
  with_env(...) do
    output = shell_output(...)
  end
  assert_match "Completed (1 file worked, 1 file failed)", output
end

Reply:
Good catch. Will fix this.